controllers/provisioning_controller.go

```diff
@@ -365,13 +386,35 @@ func (r *ProvisioningReconciler) deleteMetal3Resources(info *provisioning.Provis
 	return nil
 }
 
+func apiServerHost() (string, error) {
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return "", err
+	}
+
+	apiServerURL, err := url.Parse(config.Host)
+	if err != nil {
+		return "", err
+	}
+	host := config.Host
+	if strings.Contains(apiServerURL.Host, ":") {
+		host = strings.Split(apiServerURL.Host, ":")[0]
+	}
```

Comment:
The IPv6 failures were real, and this is the source.  Core operators like ingress that fail often indicate workers didn't deploy, and workers didn't deploy since CBO didn't bring up Metal3. 

CBO logs:

``E0604 06:00:58.813292       1 controller.go:267] controller-runtime/manager/controller/provisioning "msg"="Reconciler error" "error"="lookup [fd02: no such host" "name"="provisioning-configuration" "namespace"="" "reconciler group"="metal3.io" "reconciler kind"="Provisioning" ``

The API host might be something like `[fd02::1]:6443` and this code is splitting it -- I think this is trying to remove the port? You'll get `[fd02` instead.

Here's an example using `SplitHostPort` that should work for everything: https://play.golang.org/p/6VqhuZQmtGc